src/modules/language/language.service.ts

```diff
@@ -0,0 +1,71 @@
+import { HttpException, HttpStatus, Injectable, Logger } from '@nestjs/common';
+import { InjectRepository } from '@nestjs/typeorm';
+import { Repository } from 'typeorm';
+import { Language } from '../language/entities/language.entity';
+import { CreateLanguageDto } from './dto/create-language.dto';
+import {
+  ERROR_OCCURED,
+  LANGUAGE_ALREADY_EXISTS,
+  LANGUAGE_CREATED_SUCCESSFULLY,
+  SUCCESS,
+  FETCH_LANGUAGE_FAILURE,
+} from '../../helpers/SystemMessages';
+
+@Injectable()
+export class LanguageService {
+  constructor(
+    @InjectRepository(Language)
+    private readonly languageRepository: Repository<Language>
+  ) {}
+
+  async createLanguage(createLanguageDto: CreateLanguageDto): Promise<any> {
+    try {
+      const languageExists = await this.languageRepository.findOne({ where: { code: createLanguageDto.code } });
+
+      if (languageExists) {
+        return {
+          status_code: HttpStatus.BAD_REQUEST,
```

Comment:
409 status code is more appropriate. also known as conflict when trying to create a resource that already exists. use HttpStatus.CONFLICT instead